src/Yoti.Auth/ActivityDetail.cs

```diff
@@ -12,20 +12,42 @@ public enum ActivityOutcome { Success, ProfileNotFound, Failure, SharingFailure
     /// </summary>
     public class ActivityDetails
     {
+        public ActivityDetails(ActivityOutcome activityOutcome)
+        {
+            UserProfile = null;
+            Profile = null;
+            ReceiptID = null;
+            Outcome = activityOutcome;
+        }
+
+        public ActivityDetails(YotiUserProfile yotiUserProfile, YotiProfile yotiProfile, string receiptID, ActivityOutcome activityOutcome)
+        {
+            UserProfile = yotiUserProfile;
+            Profile = yotiProfile;
+            ReceiptID = receiptID;
+            Outcome = activityOutcome;
+        }
+
         /// <summary>
         /// The <see cref="YotiUserProfile"/> returned by Yoti if the request was successful.
         /// </summary>
         [Obsolete("Please use Profile instead")]
-        public YotiUserProfile UserProfile { get; set; }
+        public YotiUserProfile UserProfile { get; private set; }
 
         /// <summary>
         /// The <see cref="YotiProfile"/> returned by Yoti if the request was successful.
         /// </summary>
-        public YotiProfile Profile { get; set; }
+        public YotiProfile Profile { get; private set; }
 
         /// <summary>
         /// The outcome status of the request.
         /// </summary>
-        public ActivityOutcome Outcome { get; set; }
+        public ActivityOutcome Outcome { get; private set; }
```

Comment:
Technically, although no-one should be using this, it is a breaking change isn't it

Reply:
Yes you're right, I'll have to delay making these private until v3.0.0